frontend/pluginloader.lua

```diff
@@ -20,6 +21,22 @@ local DEPRECATION_MESSAGES = {
     feature = _("The following features are unmaintained and will be removed soon:"),
 }
 
+local function isProvider(name)
+    return name:sub(1, 8) == "provider"
```

Comment:
provider or addon ?

`provider-samba.koplugin` vs `addon-samba.koplugin`?

Reply:
Provider seems better? Addon sounds like a synonym for extension.